src/bindings/python/src/pyopenvino/core/common.hpp

```diff
@@ -84,6 +84,19 @@ py::array array_from_tensor(ov::Tensor&& t, bool is_shared);
 
 }; // namespace array_helpers
 
+// Helpers for shapes
+namespace shape_helpers {
+
+template <typename T>
+void get_slice(const T& shape, size_t& start, size_t& step, size_t& slicelength, T& result) {
```

Comment:
```suggestion
void get_slice(T& result, const T& shape, const size_t start, const size_t step, const size_t slicelength) {
```
Passing built-in types as reference is not as performant as [passing them by value](https://stackoverflow.com/questions/3009543/passing-integers-as-constant-references-versus-copying). I also added some `const`s, `start` passed by value is not actually being changed in the function since it's being copied, but I'm not certain if compiler will allow marking it as `const`. If it doesn't let's leave `start` non-const.

Also, since `result` is not the only non-const argument, I'd put it first.

Reply:
You're absolutely right, thank you for pointing that out. I'll apply the suggestion right away.

> but I'm not certain if compiler will allow marking it as `const`. If it doesn't let's leave `start` non-const.

I tried locally and indeed the compiler does not allow using `const size_t start`, I will leave it non-constant.